README.md

```diff
@@ -38,7 +38,11 @@ You should also specify settings that will be shared across all the plugin rules
     "react": {
       "createClass": "createReactClass", // Regex for Component Factory to use,
                                          // default to "createReactClass"
-      "pragma": "React",  // Pragma to use, default to "React"
+      "localName": "React" // Local name of React import, default to "<setttings.react.pragma>"
+      "jsxPragma": "React.createElement" // JSX Pragma to use, default to "<settings.react.pragma>.createElement"
+      "jsxFragPragma": "React.Fragment" // JSX Fragment Pragma to use, default to "<settings.react.pragma>.<settings.react.fragment>"
+      "pragma": "React",  // *Deprecated* Pragma to use, default to "React"
```

Comment:
i'm confused, why would this be deprecated?

Reply:
Since semantics changed, I think new settings should be provided.
If you think just breaking those (and probably update major version) is better, I will update this PR.

Reply:
How have the semantics changed (for users who haven't added the new pragma settings)?

Reply:
I mean, if old users use `pragma` with `React`, originally rules will detect `<pragma>.createElement`. With `jsxPragma`, rules detect `<jsxPragma>` so that its behavior aligns with babel's.
Therefore, if I just use `pragma` and do not add new settings, it will break existing settings.

Reply:
Right but why would `jsxPragma` even exist in the common case for users of this plugin, if it doesn't do anything for them?

Reply:
@ljharb Could you tell me your suggestion for the setting combination?
I don't get the right point I need to explain, so I think that would be easier to me.

Reply:
@ljharb FYI, what I considered are
1. A case with custom element creator
    ```
    import { jsx } from 'my-jsx';
    import React from 'react';

    <><div /></>
    ```
1. A case with different local name for React
    ```
    import R from 'react';

    <><div /></>
    ```
1. A case with different library with Fragment support.
    ```
    import O from 'other-fragment-support-jsx-library';

    <><div /></>
    ```
1. Cases with mixes of others.

Reply:
OK, so. To support your first 3 cases:
1. currently not supported - `createElement` is hardcoded off of the `pragma`.
2. currently uses `pragma`; `localName` is fine to also support via an inline comment, but `pragma` should remain the only way to configure this in settings.
3. currently uses `fragment` in `getFragmentFromContext`, and then pairs that [here](https://github.com/yannickcr/eslint-plugin-react/blob/dd0757fbf7ab96770282a2a64c9b696856a47235/lib/rules/jsx-fragments.js#L43-L44) with `pragma` - if `jsxFragPragma` should override both, then that's fine to support as an inline comment, and maybe we could make the `fragment` setting be either a string (paired with pragma), or an object where either is supported?
4. without specifics i'm not sure what we need to do here.

It seems like we should maybe do these new features as separate PRs.

The first one seems to be 3 - ie, add support for an alternative schema in `fragment`, that inline `jsxFrag` comments hook into.